src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -601,6 +602,19 @@ public sealed override ImmutableArray<FieldSymbol> TupleElements
                                 continue;
                             }
 
+                            if (this.IsDefinition)
+                            {
+                                // No need to wrap fields on System.ValueTuple definitions
+                                members.Add(field);
+                                int tupleFieldIndex2 = currentFieldsForElements.IndexOf(field, ReferenceEqualityComparer.Instance);
```

Comment:
>currentFieldsForElements [](start = 55, length = 24)

Consider using a `HashSet` for `currentFieldsForElements`.

Reply:
Never mind. This is the same lookup we were performing previously, and there are at most 8 items in the collection.

---
In reply to: [425355278](https://github.com/dotnet/roslyn/pull/44231#discussion_r425355278) [](ancestors = 425355278)